### ISSUE TITLE
Tile mode needs refactoring

### ISSUE THREAD

**Issue by Abalieno** [CleverRaven/Cataclysm-DDA#4236]
I explained this on the forums. Right now it's just not possible to complete a tileset and maintain it. You just need a million of hours and every time stuff is added you have to go through a lot of busywork.

Here's the suggested refactoring to make this work:
## 

I just spent a VERY long time going through all "books.json" to add ALL entries to the tileset. It's insane work.

This is something that needs to be refactored and that is more important than categories: we need a way to bind tiles NOT through IDs, but through ASCII+color.

It's just not remotely possible to bind every single ID for the billion of objects in the game. The real problem here is that there are like 50 books that have different IDs but that are all shown as a pink "?". So I don't want to use 50 entries IDs in my tiles file pointing to the same tile. I just want to bind the pink "?" ASCII to a tile. Once. All objects that are shown as a pink "?" will then default to that tile.

So, basically, we should build tiles for all used ASCII+color combinations. There should be an automatic method/output file that lists every combination and what it is associated to. Something like the green "/" represents this list of objects. And then let tile-makers bind one tile to the green "/", and not IDs (unless overridden with a specific ID).

So the "tile_config.json" should first have a mandatory block where you bind all ASCII and colors in use (surely a MUCH SMALLER number than all the IDs in the game). THEN an optional block where you bind single IDs, if you want.

That way every time you add stuff to the game, you ALWAYS have tile mode working perfect, because it always knows how to output that ASCII+color combination by default.
##

**Comment by illi-kun** [comment]
### This issue was closed as it appears inactive.

Reducing open issues to those which are (or will) be actively worked upon helps us focus our efforts. This issue has not been deleted - it still appears in searches and if it contains relevant information you are encouraged to continue to link to it.

### If this issue was a bug
It should be reopened if it can be reproduced in the current build. You can obtain the most recent copy [here](http://dev.narc.ro/cataclysm/jenkins-latest/). Please check there is not a more recent report of this bug before doing so. If no more recent report exists you should continue the discussion in this issue.

### If this was a feature request
If the consensus was that the idea was good you could consider submitting an implementation via a PR. If you want to comment further please do so here as opposed to opening a new issue. Before posting check nobody has already made the same point and consider whether your comments are likely to lead to an implementation. If you have doubts about either consider instead [voting](http://iv.cataclysmdda.com:8111/all) for the issue

### If you want to work on this issue
Then either assign it to yourself or if you are unable to do so claim it via adding a comment. Please don't assign others or make a general request for action.
